rtengine/diagonalcurves.cc

```diff
@@ -106,18 +106,18 @@ DiagonalCurve::DiagonalCurve (const std::vector<double>& p, int poly_pn)
                 }
 
                 for (int i = 4; i < 8; i++) {
-                    x[i] = (p[i] + 100.0) / 200.0;
+                    x[i] = max((p[i] + 100.0) / 200.0, 0.001);
```

Comment:
If I understand the whole chain of events properly, I don't think this is an acceptable change from a backward compatibility point of view: people that have previously set their sliders to -100 would get a different result when reprocessing the file.
Also: only adding a margin at the bottom seems arbitrary, because I can also get the artifacts you try to solve when setting Lights -> 100 and pushing the third locus to the far right.